File: 02_ProgrammingFundamentals/_11_RegularExpressions/_01_Lecture/_01_MatchFullName.java
package _02_ProgrammingFundamentals._11_RegularExpressions._01_Lecture;

import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class _01_MatchFullName {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        
        // Read input value
        String console = scan.nextLine();
        
        // Add variables
        String regexName = "(?<name>\\b[A-Z][a-z]+ [A-Z][a-z]+\\b)";
        Pattern patternName = Pattern.compile(regexName);
        Matcher matcherName = patternName.matcher(console);
        
        // Match the name and print result
        while (matcherName.find()) {
            String find = matcherName.group("name");
            
            System.out.print(find + " ");
        }
        
        scan.close();
    }
}

File: 24_C#Fundamentals/_09_RegularExpressions/_01_Lecture/_01_MatchFullName/_01_MatchFullName.cs
using System;
using System.Text.RegularExpressions;

public class _01_MatchFullName
{
    public static void Main(string[] args)
    {
        // Read input value
        string console = Console.ReadLine();

        // Add variables
        string regexName = "(?<name>\\b[A-Z][a-z]+ [A-Z][a-z]+\\b)";
        Regex patternName = new Regex(regexName);
        Match matcherName = patternName.Match(console);

        // Match the name and print result
        while (matcherName.Success)
        {
            string find = matcherName.Groups["name"].Value;
            Console.Write(find + " ");

            matcherName = matcherName.NextMatch();
        }
    }
}
